lib/bpf/bpf.go

```diff
@@ -171,19 +171,30 @@ func New(config *Config) (BPF, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	s.conn, err = startConn(*config.NetworkBufferSize)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
 
-	log.Debugf("Started enhanced session recording with buffer sizes (command=%v, "+
-		"disk=%v, network=%v) and cgroup mount path: %v. Took %v.",
-		*s.CommandBufferSize, *s.DiskBufferSize, *s.NetworkBufferSize, s.CgroupPath,
-		time.Since(start))
+	if restrictedSession.Enabled {
+		// Load network BPF modules only when required.
+		s.conn, err = startConn(*config.NetworkBufferSize)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		log.Debugf("Started enhanced session recording with buffer sizes (command=%v, "+
+			"disk=%v, network=%v) and cgroup mount path: %v. Took %v.",
+			*s.CommandBufferSize, *s.DiskBufferSize, *s.NetworkBufferSize, s.CgroupPath,
+			time.Since(start))
+
+		go s.processNetworkEvents()
```

Comment:
any risk of race conditions in those `s.emitXXX` functions now we have two separate event loops? For example, we do `s.argsCache.Get` then `s.argsCache.Set`, probably ok...